### ISSUE TITLE
Unusual Network Connection via RunDLL32

### ISSUE THREAD

**Issue by garyjblackwell** [elastic/detection-rules#518]
The detection rule for when rundll32.exe makes a network connection alerts when Logitech requests an update.  The IP addresses for Logitech are hosted on cloudflare so there are a lot and not a specific range.  

Here is the query:
signal.rule.query: "process.name:rundll32.exe and event.action:"Network connection detected (rule: NetworkConnect)" and not destination.ip:10.0.0.0/8 and not destination.ip:172.16.0.0/12 and not destination.ip:192.168.0.0/16"

We need a way to exclude rundll32.exe network requests to Logitech, while still getting other suspicious network requests.

**Comment by peasead** [comment]
Thanks for opening the Issue, and good question.

Could you drop a JSON output from one of the events from Discover (make sure you remove any sensitive information from it)? There may be process fields that we can use to help you tune the rule to your environment. 

![image](https://user-images.githubusercontent.com/7442091/98737471-d9f33680-236b-11eb-91b1-45b53e691144.png)

**Comment by garyjblackwell** [comment]
Here is the JSON
[json-for-Detection.txt](https://github.com/elastic/detection-rules/files/5520437/json-for-Detection.txt)

**Comment by garyjblackwell** [comment]
[json_nearby_Discover_Evnt.txt](https://github.com/elastic/detection-rules/files/5520469/json_nearby_Discover_Evnt.txt)

**Comment by peasead** [comment]
Thanks @garyjblackwell 

We're taking a look at those logs and will circle back with you.

**Comment by peasead** [comment]
That rule is being converted into an EQL rule to allow for better sequencing, specifically `rundll32` with no command-line arguments and a network connection. This should be released in `7.10` and you can preview it [here](https://github.com/elastic/detection-rules/blob/7.10/rules/windows/execution_unusual_network_connection_via_rundll32.toml) or even preemptively load it using the `detection_rules` [Python module](https://github.com/elastic/detection-rules/blob/main/CLI.md#uploading-rules-to-kibana). Note, you're on `7.5.1` and ECS `1.1.0`, so when you upgrade to `7.10`, that rule will automatically be available as EQL.

In the event upgrading to the current version isn't an option, you can use the exception framework that is built into the Security App.

<img width="431" alt="Detections_-_Kibana" src="https://user-images.githubusercontent.com/7442091/98843142-2bed9800-2410-11eb-9280-d7b7319b832a.png">

<img width="985" alt="Detections_-_Kibana" src="https://user-images.githubusercontent.com/7442091/98843769-01e8a580-2411-11eb-9985-e2fd41c57691.png">

**Comment by garyjblackwell** [comment]
I put in a request for our security engineering to upgrade our instance.  Can you share an example of how the exception would work when it is an EQL one?

**Comment by peasead** [comment]
@garyjblackwell on the left side of the image, you'll see that there is an `AND` operator between the two fields meaning that for the exception to apply, you'd first need the rule to fire and then the exception is evaluated to ensure the `destination.domain` is `*.cloudfront.net` **and** the `user.name` is `SYSTEM`. Only if all three of those elements exist (rule, destination, and user), will the exception occur.

Ah. It looks like the [Exception Framework](https://www.elastic.co/blog/whats-new-elastic-security-7-9-0-free-endpoint-security) was released in `7.9`.

All that aside, I think your best option would be to update to the lastest version of the Elastic Stack (sounds like you've kicked that process off), and until then, disable the rule and write you own that incorporates the exception logic. Something like
```
process.name:rundll32.exe and event.action:"Network connection detected (rule: NetworkConnect)" and not 
destination.ip:10.0.0.0/8 and not destination.ip:172.16.0.0/12 and not destination.ip:192.168.0.0/16 and not 
(destination.domain:*.cloudflare.net and user.name:SYSTEM)
```

**Comment by garyjblackwell** [comment]
I added the exception, and it has been working.
